portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -60,15 +84,58 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     // Get the input from the form.
     String userComment = request.getParameter("user-comment");
     String userName = request.getParameter("user-name");
+    String userMood = request.getParameter("mood");
+    Long timestamp = System.currentTimeMillis();
+
+    userFilter = getFilterChoice(request);
+    commentNumber = getNumberComment(request);
+    if (commentNumber == -1) {
+      response.setContentType("text/html");
+      response.getWriter().println("Please enter an integer between 0 and 10.");
+      return;
+    }
 
     Entity commentEntity = new Entity("Comment");
     commentEntity.setProperty("text", userComment);
     commentEntity.setProperty("author", userName);
+    commentEntity.setProperty("mood", userMood);
+    commentEntity.setProperty("timestamp", timestamp);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     // Redirect back to the HTML page.
     response.sendRedirect("/comment.html");
   }
+
+  public String getFilterChoice(HttpServletRequest request) {
```

Comment:
These methods can be `static`. This keyword helps me understand that these methods do not modify state of instances of DataServlet.